src/Tribe/Cache.php

```diff
@@ -163,8 +166,14 @@ public function delete_transient( $id, $expiration_trigger = '' ) {
 	 * This uses a modification of the the query from https://core.trac.wordpress.org/ticket/20316
 	 *
 	 * @since 4.11.0
+	 *
+	 * @return void Just execute the database SQL no return required.
 	 */
 	public function delete_expired_transients() {
+		if ( tribe_get_var( 'has_deleted_expired_transients', false ) ) {
```

Comment:
Why use the global container here? 
If this class is the only one in charge of handling transient deletions, then a static var should do.

Is there any specific reason to use `tribe_(get|set)_var`?

If the idea is to allow external, third-party code, to handle this and then set `tribe_set_var( 'has_deleted_expired_transients' )`, then this should be documented in this method.

I would add a filter too, here, to bail out of the deletion entirely:

```
$delete_expired_transients = apply_filters( 'tribe_cache_delete_expired_transients', (bool) static::$transients_deleted );
if( false === (bool)$delete_expired_transients ) {
    return;
}

// ... here do the deletion

static::$transients_deleted = true;
```

Reply:
The problem here is that you can have multiple instances of the class which the same class which was one of the big problems of the previous implementation.